app/Services/RegisterService.php

```diff
@@ -0,0 +1,36 @@
+<?php
+namespace App\Services;
+
+use App\User;
+use Illuminate\Support\Facades\Hash;
+use App\Models\UserProfile;
+use App\Contracts\RegisterContract;
+use function Opis\Closure\unserialize;
+use function Opis\Closure\serialize;
+
+
+class RegisterService implements RegisterContract
+{
+    public function register($request)
+    {
+        try {
+            $name = (string)$request['name'];
+            $email = (string)$request['email'];
+            $password = (string)$request['password'];
+        } catch (\Exception $e) {
+            return ['message_error' => 'User was not successfully created.'];
+        }
+
+        //TODO: add organization destinction
+        try {
```

Comment:
There's a createOrFail function or you can surround this in a database transaction -> https://laravel.com/docs/5.8/database#database-transactions